.travis.yml

```diff
@@ -1,15 +1,106 @@
-# Config file for automatic testing at travis-ci.org
-
-sudo: required
-
-language: python
+# test stage will test all branches against:
+#   - linux python 3.5, 3.6, 3.7, 3.8
+#   - osx python 3.7
+#
+# deploy stage builds and test the wheels when jobs is
+#   - pull request
+#   - master branch
+#   - tagged commit, only this will be uploaded to pypi
 
 services:
   - docker
 
-before_install:
-  - docker build --build-arg REQUIREMENTS=_dev -t usdaarsnwrc/smrf .
+stages:
+  - test
+  - name: deploy
+    if:  type = pull_request OR branch = master OR tag IS present AND repo = USDA-ARS-NWRC/smrf
```

Comment:
The deploy doesn't work for SMRF and it has to be a bit more manual. It's using `cibuildwheel` to create the wheels to upload to pypi and wouldn't work for the built in Travis pypi deployment stage.

This will also trigger the deploy build on all pull requests and master branches to ensure that the wheels can be built on all supported OS's. It will only deploy to pypi if the tag is present (in `travis/build.sh`).